src/jdk.incubator.code/share/classes/jdk/incubator/code/parser/impl/DescParser.java

```diff
@@ -51,190 +45,49 @@ public static TypeElement.ExternalizedTypeElement parseExTypeElem(String desc) {
         return parseExTypeElem(s);
     }
 
-    /**
-     * Parse a method reference from its serialized textual form.
-     *
-     * @param desc the serialized method reference
-     * @return the method reference
-     */
-    public static MethodRef parseMethodRef(String desc) {
-        Scanner s = Scanner.factory().newScanner(desc);
-        s.nextToken();
-        return parseMethodRef(s);
-    }
-
-    /**
-     * Parse a constructor reference from its serialized textual form.
-     *
-     * @param desc the serialized constructor reference
-     * @return the constructor reference
-     */
-    public static ConstructorRef parseConstructorRef(String desc) {
-        Scanner s = Scanner.factory().newScanner(desc);
-        s.nextToken();
-        return parseConstructorRef(s);
-    }
-
-    /**
-     * Parse a field reference from its serialized textual form.
-     *
-     * @param desc the serialized field reference
-     * @return the field reference
-     */
-    public static FieldRef parseFieldRef(String desc) {
-        Scanner s = Scanner.factory().newScanner(desc);
-        s.nextToken();
-        return parseFieldRef(s);
-    }
-
-    /**
-     * Parse a record type reference from its serialized textual form.
-     *
-     * @param desc the serialized record type reference
-     * @return the record type reference
-     */
-    public static RecordTypeRef parseRecordTypeRef(String desc) {
-        Scanner s = Scanner.factory().newScanner(desc);
-        s.nextToken();
-        return parseRecordTypeRef(s);
-    }
-
+    //    ExType:
```

Comment:
Note that the grammar here is very simple, as we no longer have to have support for Java types in here. An externalized type form is a type identifier, followed by zero or more externalized type arguments. An identifier is a qualified identifier -- each part in the identifier can be either string literals or identifiers.